iphone/Maps/Core/Settings/MWMSettings.mm

```diff
@@ -74,24 +74,34 @@ + (void)setCompassCalibrationEnabled:(BOOL)compassCalibrationEnabled
 
 + (MWMTheme)theme
 {
+  MWMTheme theme;
   if ([MWMCarPlayService shared].isCarplayActivated) {
     UIUserInterfaceStyle style = [[MWMCarPlayService shared] interfaceStyle];
     switch (style) {
-      case UIUserInterfaceStyleLight: return MWMThemeDay;
-      case UIUserInterfaceStyleDark: return MWMThemeNight;
+      case UIUserInterfaceStyleLight: theme = MWMThemeDay;
+      case UIUserInterfaceStyleDark: theme = MWMThemeNight;
       case UIUserInterfaceStyleUnspecified: break;
     }
   }
   auto ud = NSUserDefaults.standardUserDefaults;
-  if (![ud boolForKey:kUDAutoNightModeOff])
-    return MWMThemeAuto;
-  return static_cast<MWMTheme>([ud integerForKey:kThemeMode]);
+  theme = ![ud boolForKey:kUDAutoNightModeOff] ? MWMThemeAuto : static_cast<MWMTheme>([ud integerForKey:kThemeMode]);
+  [self updateSystemUserInterfaceStyle:theme];
```

Comment:
Is it a geter that also sets/updates something? Is it expected by the caller?

Reply:
Thanks for this point!
I refactored my previous implementation an move the chunck of code which updates the `userInterfaceStyle` to the `ThemeManager` (because it it's responsibility to handle themes and styles overall).